EnglishWebRoot/school/pubbly_engine/latest/js/run/dom.js

```diff
@@ -152,6 +155,67 @@ function PubblyDom(xml, environment) {
         $("#main").addClass("transformCenterCont");
         $("#main > div").addClass("transformCenter");
 
+        function scalingYCoordinate(screenAttributes) {
+            screenAttributes.style.height = '100%';
+            //  Converting 100% to pixel
+            var totalpixel = parseFloat(window.getComputedStyle(screenAttributes).height);
+            screenAttributes.style.height = domHeight;
+            //  Calculating Scale % to increase the width of the screen
+            return (totalpixel / info.height);
+        }
+
+        function scalingXCoordinate(screenAttributes) {
+            screenAttributes.style.width = '100%';
+            //  Converting 100% to pixel
+            var totalpixel = parseFloat(window.getComputedStyle(screenAttributes).width);
+            screenAttributes.style.width = domWidth;
+            //  Calculating Scale % to increase the width of the screen
+            return (totalpixel / info.width);
+        }
+
+        window.setTimeout(function (screenAttributes) {
+            var scalingFactor = 0, translatingFactor = 0;
+
+            screenAttributes = document.getElementsByClassName('transformCenter')[0];

```

Comment:
This method can be written better.
Why do you need the set Timeout?

Reply:
Refactored the code.
setTimeout() function is required to correctly fetch the viewport properties. Tested after removing the property.